blockly/src/server/Server.java

```diff
@@ -1154,21 +1154,45 @@ else if (argument.equals("true") || argument.equals("false")) {
   }
 
   /**
-   * Move the hero in a specific direction.
+   * Move the hero in a specific direction. One Move equals one Tile
    *
    * @param direction Direction in which the hero will be moved.
    */
   public void move(final Direction direction) {
+    double distanceThreshold = 0.1;
+    Entity hero = Game.hero().orElseThrow(MissingHeroException::new);
+    PositionComponent hpc =
+        hero.fetch(PositionComponent.class)
+            .orElseThrow(() -> MissingComponentException.build(hero, PositionComponent.class));
+
+    Coordinate targetedPosition =
+        hpc.position().toCoordinate().add(new Coordinate(direction.x(), direction.y()));
+    Tile targetedTile = Game.tileAT(targetedPosition);
+    if (targetedTile == null) return;
+    if (!targetedTile.isAccessible()) return;
+
     VelocityComponent vc =
         hero.fetch(VelocityComponent.class)
             .orElseThrow(() -> MissingComponentException.build(hero, VelocityComponent.class));
-    switch (direction) {
-      case UP -> vc.currentYVelocity(1 * vc.yVelocity());
-      case DOWN -> vc.currentYVelocity(-1 * vc.yVelocity());
-      case LEFT -> vc.currentXVelocity(-1 * vc.xVelocity());
-      case RIGHT -> vc.currentXVelocity(1 * vc.xVelocity());
+
+    // placing the Hero in the middle of the current tile before moving
+    hpc.position(hpc.position().toCoordinate().toCenteredPoint());
+
+    double distance = hpc.position().distance(targetedPosition.toCenteredPoint());
+    double distanceLast;
+    // moves the Hero towards the targeted center Point
+    while (true) {
+      vc.currentXVelocity(direction.x() * vc.xVelocity());
+      vc.currentYVelocity(direction.y() * vc.yVelocity());
+      distanceLast = distance;
+      distance = hpc.position().distance(targetedPosition.toCenteredPoint());
+      // check if the hero is close enough to the center or already over
+      if (!(distance > distanceThreshold && distanceLast >= distance)) break;
```

Comment:
Meinst du so? 
```suggestion
    if (distance <= distanceThreshold || distanceLast < distance) break;
```

Reply:
ja :)

Reply:
@Tobi511 ich würde sogar noch hingehen, und die letzte bedingung in der reihenfolge tauschen: `if (distance <= distanceThreshold || distance > distanceLast) break;`. 

also wie üblich links die variable, deren wert gefragt/genutzt wird (ist-wert), und auf der rechten seite der soll-wert.